cmd/docker-proxy/udp_proxy_linux.go

```diff
@@ -87,18 +114,24 @@ func (proxy *UDPProxy) replyLoop(proxyConn *net.UDPConn, serverAddr net.IP, clie
 
 	readBuf := make([]byte, UDPBufSize)
 	for {
-		proxyConn.SetReadDeadline(time.Now().Add(UDPConnTrackTimeout))
+		cte.conn.SetReadDeadline(time.Now().Add(proxy.connTrackTimeout))
 	again:
-		read, err := proxyConn.Read(readBuf)
+		read, err := cte.conn.Read(readBuf)
 		if err != nil {
 			if err, ok := err.(*net.OpError); ok && err.Err == syscall.ECONNREFUSED {
 				// This will happen if the last write failed
 				// (e.g: nothing is actually listening on the
 				// proxied port on the container), ignore it
-				// and continue until UDPConnTrackTimeout
+				// and continue until DefaultConnTrackTimeout
 				// expires:
 				goto again
 			}
+			// If the UDP connection is one-sided (i.e. the backend never sends
+			// replies), the connTrackEntry should not be GC'd until no writes
+			// happen for proxy.connTrackTimeout.
+			if errors.Is(err, os.ErrDeadlineExceeded) && cte.lastWrite() < proxy.connTrackTimeout {
```

Comment:
big nit: `lastWrite` sounds more like it would return a timestamp instead of duration.

Moving the `time.Since` out of the `lastWrite` would read a little bit nicer:
```suggestion
			if errors.Is(err, os.ErrDeadlineExceeded) && time.Since(cte.lastWrite()) < proxy.connTrackTimeout {
```